workspaces/rbac/plugins/rbac/src/components/RbacPage.tsx

```diff
@@ -16,26 +16,35 @@
 import React from 'react';
 
 import { Content, Header, Page } from '@backstage/core-components';
-import { RequirePermission } from '@backstage/plugin-permission-react';
 
 import { DeleteDialogContextProvider } from '@janus-idp/shared-react';
 
-import { policyEntityReadPermission } from '@backstage-community/plugin-rbac-common';
-
 import { RolesList } from './RolesList/RolesList';
+import { useApi } from '@backstage/core-plugin-api';
+import { rbacApiRef } from '../api/RBACBackendClient';
+import { useAsync } from 'react-use';
+import { ErrorPage } from '@backstage/core-components';
+
+export const RbacPage = ({ useHeader = true }: { useHeader?: boolean }) => {
+  const rbacApi = useApi(rbacApiRef);
+  const { loading: isUserLoading, value: result } = useAsync(
+    async () => await rbacApi.getUserAuthorization(),
+    [],
+  );
 
-export const RbacPage = ({ useHeader = true }: { useHeader?: boolean }) => (
-  <RequirePermission
-    permission={policyEntityReadPermission}
-    resourceRef={policyEntityReadPermission.resourceType}
-  >
-    <Page themeId="tool">
-      {useHeader && <Header title="RBAC" />}
-      <Content>
-        <DeleteDialogContextProvider>
-          <RolesList />
-        </DeleteDialogContextProvider>
-      </Content>
-    </Page>
-  </RequirePermission>
-);
+  if (!isUserLoading) {
+    return result?.status === 'Authorized' ? (
+      <Page themeId="tool">
+        {useHeader && <Header title="RBAC" />}
+        <Content>
+          <DeleteDialogContextProvider>
+            <RolesList />
+          </DeleteDialogContextProvider>
+        </Content>
+      </Page>
+    ) : (
+      <ErrorPage statusMessage="Not Found" />
+    );
+  }
+  return null;
```

Comment:
Should we show a page level loader here instead of returning null?

Reply:
Maybe something like the [Progress](https://backstage.io/storybook/?path=/story/feedback-progress--progress)?

Reply:
Makes sense. @ShiranHi wdyt?

Reply:
Went ahead and made the change since it was fairly small and easy to reverse.